async.lua

```diff
@@ -0,0 +1,26 @@
+local areas = {}
+local core = core
+
+local safe_file_write = core.safe_file_write
+if safe_file_write == nil then
+    function safe_file_write(path, content)
+        local file, err = io.open(path, "w")
+        if err then
+            return err
+        end
+        file:write(content)
+        file:close()
+    end
+end
+
+-- Save the areas table to a file
+function areas:do_save(areas_tb, filename)
```

Comment:
`self` is not used here, hence this function could be named `areas.do_save` or `areas._internal_do_save` such that modders are more unlikely to use the function when they shouldn't.

That also gets rid of the `async_func` wrapper function.